### ISSUE TITLE
SQLAlchemyBackend postgresql initialize error

### ISSUE THREAD

**Issue by motord** [FedericoCeratto/bottle-cork#64]
For db_full_url like 'postgresql://user:password@localhost/database', initialization will fail.

```
        if initialize:
            # Create new database if needed.
            db_url, db_name = db_full_url.rsplit('/', 1)
            self._engine = create_engine(db_url)
            try:
                self._engine.execute("CREATE DATABASE %s" % db_name)
            except Exception, e:
                log.info("Failed DB creation: %s" % e)

            # SQLite in-memory database URL: "sqlite://:memory:"
            if db_name != ':memory:':
                self._engine.execute("USE %s" % db_name)
```

**Comment by astubbs** [comment]
Whipped up some commits based on the above, let me know what you think: https://github.com/astubbs/bottle-cork/commit/ee6da497966e0d4818c2bf8d590e9838692d4313

**Comment by astubbs** [comment]
The tests don't seem to be self contained in tox. What's the process for getting this tested properly?

**Comment by astubbs** [comment]
Ignore that sha, the branch is here: https://github.com/astubbs/bottle-cork/tree/postgres-fix

**Comment by jlutz777** [comment]
I didn't test it, but your branch doesn't look quite right.  You are calling create_engine twice for postgresql and not setting the isolation level back.  It looks like a copy-paste mistake.

**Comment by astubbs** [comment]
No, I added it on purpose. How can you create the database, and connect to it - I understood you have to connect directly to the db, so you'd need to reconnect, no?
But yes you were right about turning the transactions back on, I've put that back in.
Do you know if the tests should all pass in origin/master by just simply running tox?

**Comment by jlutz777** [comment]
I see what you did now, sorry about that.  As far as the tests, I'm not sure.  As you can guess since I posted this a while ago, I didn't get around to actually implementing it and figuring out the tests.  Maybe Federico can help with some direction on running the tests.

**Comment by botzill** [comment]
Hi.

I wanted to know what is the status of this issue? Was it fixed/merged? I see there is some fix on it.

Thx.
